core/src/main/java/org/springframework/security/core/annotation/ExpressionTemplateValueProvider.java

```diff
@@ -0,0 +1,35 @@
+package org.springframework.security.core.annotation;
+
+/**
+ * Provides a mechanism for providing custom values from enum types used in security
+ * meta-annotation expressions. For example:
+ *
+ * <pre>
+ * enum Permission implements ExpressionTemplateValueProvider {
+ *   READ,
+ *   WRITE;
+ *
+ *   &#64;Override
+ *   public String getExpressionTemplateValue() {
+ *     return switch (this) {
+ *       case READ -> "user.permission-read";
```

Comment:
While not a guarantee, I imagine that most folks will want to inject a value as a string, which will mean applying single quotes around it. Because it won't work to do `'{permissions}'` in an expression, it seems preferable to always surround the value in single quotes, thus allowing a template to do `{permissions}` or `{permission}` and still have the template compile.

If you agree with that, will you please place single quotes around each returned value?

Reply:
Actually, our use case that drove me to make this PR involves producing SpEL type references to the enum so there are cases where the template value will not be a string.

Reply:
Interesting, okay thank you.